alma/alma-installments-prestashop
# Fix conflict with product details pane

### Reason for change

The insurance integration conflicts with PrestaShop's product details pane on product pages, and prevents its display.

Fixes [ECOM-2155](https://linear.app/almapay/issue/ECOM-2155)

### Code changes

- Cleaned the JS global scope by moving all variables/functions into the self-executing function in `alma-product-insurance.js`
- Changed from JSON output into data attributes to JSON output into plain JavaScript objects in the global `window.AlmaInsurance` object
- Stopped trying to reload product details data from the DOM; using the data passed along the `updatedProduct` event by PrestaShop

### How to test

_As a reviewer, you are encouraged to test the PR locally._

- Load a product page for an insurance-eligible product
    👉 The insurance widget should be displayed
    👉 You should be able to switch from the product description to the product display panel
    👉 There should be no error in the console about `dataset.product` being `null`/`undefined`


### Checklist for authors and reviewers




### Non applicable
